pkg/sql/ttl/ttljob/ttljob.go

```diff
@@ -698,16 +701,25 @@ func runTTLOnRange(
 						desc.GetModificationTime().GoTime().Format(time.RFC3339),
 					)
 				}
-
 				tokens, err := deleteRateLimiter.Acquire(ctx, int64(len(deleteBatch)))
 				if err != nil {
 					return err
 				}
 				defer tokens.Consume()
 
 				start := timeutil.Now()
-				err = deleteBuilder.run(ctx, ie, txn, deleteBatch)
+				rowCount, err := deleteBuilder.run(ctx, ie, txn, deleteBatch)
+				if err != nil {
+					return err
+				}
+
 				metrics.DeleteDuration.RecordValue(int64(timeutil.Since(start)))
+				err = job.Update(ctx, txn, func(txn *kv.Txn, md jobs.JobMetadata, ju *jobs.JobUpdater) error {
+					progress := md.Progress
+					atomic.AddInt64(&progress.Details.(*jobspb.Progress_RowLevelTTL).RowLevelTTL.RowCount, int64(rowCount))
+					ju.UpdateProgress(progress)
```

Comment:
This feels like a bad idea; you could be updating progress in jobs table very rapidly; I think you should either pace it... or just not do it at all -- use metrics instead.

Reply:
i think we should just do this once at either the end of the range, or the end of the job.